### ISSUE TITLE
`jest/expect-expect`: false positive "Test has no assertions" when assertions are inside a nested `function` declaration

### ISSUE THREAD

**Issue by ahmedelgabri** [oxc-project/oxc#25635]
## Summary

`jest/expect-expect` reports a test as having no assertions when its `expect()` calls live inside a nested **`function` declaration** defined in the test body and then called. oxlint's assertion search recurses into nested **arrow-function expressions** (those are detected correctly) but not into nested **function declarations**, so it misses the assertions and emits a false "Test has no assertions". `eslint-plugin-jest` finds assertions anywhere in the test subtree.

## oxlint version

1.78.0; originally observed on 1.69.0, re-verified on 1.72.0 and 1.78.0.

## Reproduction

`.oxlintrc.json`:

```json
{
	"plugins": ["jest"],
	"categories": { "correctness": "off" },
	"rules": { "jest/expect-expect": "error" }
}
```

`b.test.ts`:

```ts
describe('suite', () => {
	// assertion in a nested FUNCTION DECLARATION → oxlint reports "no assertions" (false positive)
	test('via function declaration', async () => {
		async function check() {
			expect(1 + 1).toBe(2)
		}
		await check()
	})

	// same, but a nested ARROW FUNCTION → oxlint correctly finds the assertion (no report)
	test('via arrow function', async () => {
		const check = async () => {
			expect(1 + 1).toBe(2)
		}
		await check()
	})
})
```

```sh
oxlint -c .oxlintrc.json b.test.ts
```

## Expected

No diagnostic for either test — both contain an `expect()` reachable from the test, consistent with `eslint-plugin-jest` (no error on either test across ESLint 7 and ESLint 9 setups).

## Actual

```
b.test.ts:3:2: error jest(expect-expect): Test has no assertions help: Add assertion(s) in this Test
```

Only the function-declaration test is flagged; the arrow-function test is not — both contain the same assertion.

## Cross-checked

| engine | result |
| --- | --- |
| `eslint-plugin-jest` 27.6.0 (ESLint 7.23.0) | no error |
| `eslint-plugin-jest` 29.15.4 (ESLint 9.39.4, `typescript-eslint` 8.62.1 parser) | no error |
| oxlint 1.78.0 (also 1.69.0 / 1.70.0 / 1.72.0) | **error** (function-declaration case only) |

The asymmetry (arrow-function helper detected, `function`-declaration helper not) points to the assertion-collection walk not descending into `FunctionDeclaration` nodes.
